lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -166,12 +172,21 @@ def __init__(self, handle, *, ignore_lock=False):
         super().__init__(handle, ignore_lock=ignore_lock)
 
 
+class _ApplyProfileTemplatesOpenSearch(EventBase):
```

Comment:
This event is needed as we have to add profile templates and the cluster may not be ready. This custom events allows us to execute these tasks when the cluster is set.